pyiron_gui/project/project_browser.py

```diff
@@ -99,13 +105,138 @@ def __repr__(self):
+            select_atoms=None,
+            background="white",
+            color_scheme=None,
+            colors=None,
+            scalar_field=None,
+            scalar_start=None,
+            scalar_end=None,
+            scalar_cmap=None,
+            vector_field=None,
+            vector_color=None,
+            magnetic_moments=False,
+            view_plane=np.array([0, 0, 1]),
+            distance_from_camera=1.0,
+            opacity=1.0
+        )
+        if not self._options['reset_view'] and len(orient) == 16:
+            self._ngl_widget.control.orient(orient)
+        return self._ngl_widget
+
+
+class MurnaghanWrapper(BaseWrapper):
+    def __init__(self, pyi_obj, project, rel_path=""):
+        super().__init__(pyi_obj, project, rel_path=rel_path)
+        self._type = 'murnaghan'
+        self._option_widgets = None
+        self._options = {
+            "fit_type": self._wrapped_object.input['fit_type'],
+            "fit_order": 3  # self._wrapped_object.input['fit_order']
+        }
+        self._init_option_widgets()
+
+    def _init_option_widgets(self):
+        self._option_widgets = {
+            "fit_type": widgets.Dropdown(value=self._options['fit_type'],
+                                         options=['polynomial', 'birch', 'birchmurnaghan',
+                                                  'murnaghan', 'pouriertarantola', 'vinet'],
+                                         description='Fit type',
+                                         description_tooltip='Type of the energy-volume curve fit.'),
+            "fit_order": widgets.IntText(value=self._options['fit_order'],
+                                         description='Fit order',
+                                         description_tooltip="Order of the polynomial for 'polynomial' fits, "
+                                                             "ignored otherwise")
+        }
+
+    @property
+    def option_representation(self):
+        """ipywidet to change the options for the self_representation"""
+        return widgets.VBox([self._option_widgets['fit_type'], self._option_widgets['fit_order']])
+
+    def _parse_option_widgets(self):
+        for key in self._options.keys():
+            self._options[key] = self._option_widgets[key].value
+
+    def self_representation(self):
+        plt.ioff()
+        self._parse_option_widgets()
+
+        if self._options['fit_type'] == "polynomial" and (
+                self._wrapped_object.input['fit_type'] != "polynomial" or
+                self._wrapped_object.input['fit_order'] != self._options['fit_order']
+        ):
+            self._wrapped_object.fit_polynomial(fit_order=self._options["fit_order"])
+        elif self._options['fit_type'] != "polynomial" and (
+            self._wrapped_object.input['fit_type'] != self._options['fit_type']
+        ):
+            self._wrapped_object._fit_eos_general(fittype=self._options['fit_type'])
```

Comment:
why is `fit_polynomial` a public method of the wrapped object but `_fit_eos_general` private? Naively it feels like these should have the same exposure level. Also, I'm sure pycharm is nitting you that you're accessing a private method in the `_wrapped_object._fit_eos_general` case 😉

Reply:
True, the `Murnaghan` job has actually methods for each fit style, which under the hood call the `general` method with a corresponding string (except the `fit_polynomial` which is handled differently). I will extend the logic to not pass strings to the `Murnaghan`. This should stabilize the interface.